MimeKit/ParameterList.cs

```diff
@@ -896,16 +896,16 @@ static string DecodeRfc2231 (out Encoding encoding, ref Decoder decoder, HexDeco
 				if (TryGetCharset (text, ref index, endIndex, out string charset)) {
 					try {
 						encoding = CharsetUtils.GetEncoding (charset, "?");
-						decoder = (Decoder) encoding.GetDecoder ();
+						decoder = encoding.GetDecoder ();
 					} catch (NotSupportedException) {
 						encoding = Encoding.GetEncoding (28591); // iso-8859-1
-						decoder = (Decoder) encoding.GetDecoder ();
+						decoder = encoding.GetDecoder ();
 					}
 				} else {
 					// When no charset is specified, it should be safe to assume US-ASCII...
 					// but we all know what assume means, right??
 					encoding = Encoding.GetEncoding (28591); // iso-8859-1
-					decoder = (Decoder) encoding.GetDecoder ();
+					decoder = encoding.GetDecoder ();
 				}
```

Comment:
We should definitely get rid of these casts. I think they are a relic of when I had to depend on my Portable.Text.Encoding library for charset support on Windows Phone back when I was trying to support that platform.

And therefore I had a `using Decoder = Portable.Text.Encoding.Decoder;`